### PR TITLE
DO NOT MERGE: Auth proposal

### PR DESCRIPTION
**Pull request by aquamatthias**
This PR is for reviewing the idea of Authentication/Authorization.

### CODE REVIEW

File: marathon-interface/src/main/scala/mesosphere/marathon/interface/PathId.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def path: List[String]
+
+  /**
+    * Indicates, if an other path identifier matches this path identifier.
+    * E.g. PathId(A :: B :: Nil).matches(A :: Nil) shouldBe true
+    * E.g. PathId(A :: B :: Nil).matches(A :: C :: Nil) shouldBe false
+    * @param other the other path id to compare to.
+    * @return true if this path id matches otherwise false.
+    */
+  def matches(other: PathId): Boolean
```
Comment (gkleiman) [inline marathon-interface/src/main/scala/mesosphere/marathon/interface/PathId.scala]: I am not sure I really like the name of this method.

Do you think something like "includes" would be better?

File: marathon-interface/src/main/scala/mesosphere/marathon/interface/auth/Authenticator.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+  * Base trait for all authenticator implementations.
+  * Every authenticator can get data from an HTTP request and derive an Identity.
+  * The Identity usually is maintained in a separate Identity and Access Management System (e.g. LDAP)
+  *
+  * The Identity is passed to the [[Authorizer]].
+  * It should hold all information to make decisions possible for authorization.
+  *
+  * The Resources will make sure, that the [[Authenticator#authenticate]]
```
Comment (gkleiman) [inline marathon-interface/src/main/scala/mesosphere/marathon/interface/auth/Authenticator.scala]: This sentence is incomplete.

Did you want to say that the resources have to make sure that Authenticator#authenticate is called only once per request?

Comment (aquamatthias) [inline marathon-interface/src/main/scala/mesosphere/marathon/interface/auth/Authenticator.scala]: Hopefully not any longer.

**Comment by aquamatthias** [comment]
@gkleiman Please have a second look.
I went over all existing Resources and added AuthN/Z
What is missing:
- [ ] Rebase this PR -> once #1901 is merged
- [ ] Add Test Cases for that logic

**Pull request closed without merging**